kiwi/system/setup.py

```diff
@@ -1278,38 +1280,50 @@ def _import_custom_scripts(self):
             )
 
     def _call_script(
-        self, name, option_list=None, path_prefix=defaults.IMAGE_METADATA_DIR
+        self,
+        name: str,
+        option_list: List[str] = [],
+        path_prefix: str = defaults.IMAGE_METADATA_DIR,
+        root_is_mountpoint: bool = False
     ):
-        script_path = os.path.join(self.root_dir, path_prefix, name)
-        if os.path.exists(script_path):
-            options = option_list or []
-            if log.getLogFlags().get('run-scripts-in-screen'):
-                # Run scripts in a screen session if requested
-                command = ['screen', '-t', '-X', 'chroot', self.root_dir]
-            else:
-                # In standard mode run scripts without a terminal
-                # associated to them
-                command = ['chroot', self.root_dir]
-            if not Path.access(script_path, os.X_OK):
-                command.append('bash')
-            command.append(
-                os.path.join(os.sep, path_prefix, name)
-            )
-            command.extend(options)
-            profile = Profile(self.xml_state)
-            caller_environment = copy.deepcopy(os.environ)
-            caller_environment.update(profile.get_settings())
-            config_script = Command.call(command, caller_environment)
-            process = CommandProcess(
-                command=config_script, log_topic='Calling ' + name + ' script'
-            )
-            result = process.poll_and_watch()
-            if result.returncode != 0:
-                raise KiwiScriptFailed(
-                    '{0} failed: {1}'.format(name, result.stderr)
+        with ExitStack() as stack:
```

Comment:
Shouldn't we first check for the script and then setup bind mounts? In the current form, we will always bind mount root even if disk.sh is not there

Reply:
definitely yes :facepalm: I will adapt it